### PR TITLE
Add no warehouses info when working with stock quantities

### PR DESCRIPTION
**Pull request by orzechdev**
I want to merge this change because... it informs the user about not configured warehouses when working with stock quantities (in product and variant pages).

Zrzut ekranu 2020-09-22 o 13 29 43

~~Additionally, user may follow the link to the warehouse configure page, by doing one of the following scenarios:~~
- ~~making no changes on product or variant pages and clicking `configure a warehouse`, then goes directly to warehouse configure page~~
- ~~making changes on product or variant pages and clicking `configure a warehouse`, then dialog asking for saving changes is shown and then goes to warehouse configure page~~

~~To implement together saving page changes by modal and redirecting after that, I have added `nextAction` variable passed on submit or on submit skip. I have placed modal inside pages, not views, as it requires to call submit callback from pages form.~~

Save changes confirmation dialog will be implemented in separate PR.

**PR intended to be tested with API branch:**  master

### Pull Request Checklist

1. [x] All visible strings are translated with proper context.
1. [x] All data-formatting is locale-aware (dates, numbers, and so on).
1. [x] Translated strings are extracted.
1. [x] Number of API calls is optimized.
1. [x] The changes are tested.
1. [x] Data-test are added for new elements.
1. [x] Type definitions are up to date.
1. [x] Changes are mentioned in the changelog.

### Test environment config

API_URI=https://master.staging.saleor.rocks/graphql/

### CODE REVIEW

**Comment by patrys** [comment]
Task linked: [SALEOR-982 Dashboard should inform the user about not configured warehouses when working with stock quantities](https://app.clickup.com/t/2549495/SALEOR-982)

**Comment by patrys** [comment]
Task linked: [SALEOR-758 Show empty state in product form when warehouses are not configured](https://app.clickup.com/t/2549495/SALEOR-758)

**Review by AlicjaSzu — APPROVED** [review]

File: src/products/views/ProductVariantCreate.tsx [hunk 1]
```diff
@@ -142,6 +143,7 @@ export const ProductVariant: React.FC<ProductVariantCreateProps> = ({
         onBack={handleBack}
         onSubmit={handleSubmit}
         onVariantClick={handleVariantClick}
+        onWarehouseConfigure={() => navigate(warehouseListPath)}
```
Comment (dominik-zeglen) [inline src/products/views/ProductVariantCreate.tsx:146]: Shouldn't it point to warehouse creation view?

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

File: src/products/views/ProductVariantCreate.tsx (hunk 1 above)
Comment (orzechdev) [reply src/products/views/ProductVariantCreate.tsx]: yep, might be better

File: src/products/components/ProductStocks/ProductStocks.tsx [hunk 2]
```diff
@@ -191,32 +191,40 @@ const ProductStocks: React.FC<ProductStocksProps> = ({
...
+                      <Link onClick={onWarehouseConfigure}>
+                        <FormattedMessage
+                          defaultMessage="configure a warehouse"
+                          description="no warehouses info"
+                          id="productVariantWarehouseSectionDescriptionLink"
+                        />
+                      </Link>
+                    )
+                  }}
+                />
```
Comment (dominik-zeglen) [inline src/products/components/ProductStocks/ProductStocks.tsx]: Okay I guess I need to be more precise:
https://formatjs.io/docs/react-intl/components/#rich-text-formatting

So I'd expect something like this:
`defaultMessage="There are no warehouses set up for your store. To add stock quantity to the variant please <link>configure a warehouse</link>"`

**Review by dominik-zeglen — CHANGES_REQUESTED** [review]

**Review by dominik-zeglen — APPROVED** [review]

**Comment by orzechdev** [comment]
passed to QA

**Review by mlamperski — APPROVED** [review]
Tested positive

**Pull request merged**